backend/hello.py

```
@@ -28,6 +28,13 @@ def get_lobby(code: str):
         return {"error": "Lobby not found"}
     return {"code": code, "players": lobby["players"]} 
 
+@app.post("/lobby/{code}/join")
+def join_lobby(code:str, player:str):
+    lobby = lobbies.get(code)
+    if lobby is None:
+        return {"error": "Lobby not found"}
+    lobby["players"].append(player)
```

should use a try except. also raise a http exception rather than returning a success response. also, use the `Player` model

---

I agree that something like a 404 would be more descriptive for this case.
Just read into it and it looks pretty straight forward 
`@app.get("/items/{item_id}")
async def read_item(item_id: str):
    if item_id not in items:
        raise HTTPException(status_code=404, detail="Item not found")
    return {"item": items[item_id]}`
[https://fastapi.tiangolo.com/tutorial/handling-errors/](https://fastapi.tiangolo.com/tutorial/handling-errors/)

---

I added the try except and the HTTP exception. Changing player to the Player model kept bringing up errors, I will take a look at that tomorrow.

---

For the player model feedback, does this mean using a player object as a parameter for the request instead of an id?

---

Also, how will these ids be determined? The way we have it now the id is specified by the frontend which is okay if the frontend first gets the id from the backend. Otherwise we could run into users with == ids. 

I also think this ticket is covering the functionality of joining a lobby and the http endpoint for players joining a lobby. I think we need to start being more specific with our tickets and AC since this pr already does cover being able to join a lobby, but now we aren't merging it because of http endpoint stuff. 

The API end points are currently doing logic which I think we should separate, so like helper functions that get called by the endpoints instead of these endpoints methods having game logic.

---

> The API end points are currently doing logic which I think we should separate, so like helper functions that get called by the endpoints instead of these endpoints methods having game logic. This way we can implement lobby join functionality as its own thing and unit test it easily without having to use postman/call end points. Then the end points can also be tested easier since they won't contain game logic themselves just testing the fact they get hit appropriately.
agreed